src/openapi/streaming/subscription.ts

```diff
@@ -363,6 +370,15 @@ class Subscription {
             this.networkErrorSubscribingTimer = null;
         }
 
+        if (
+            this.shouldSubscribeBeforeStreamingSetup &&
+            action === ACTION_SUBSCRIBE &&
+            !this.connectionAvailable

```

Comment:
I'm a little nervous about this. Essentially we are relying on connectionAvailable - but what if you lose connection - do we want a resubscribe to happen straight away? probably not.
I think we need a solution where we will literally only try to early subscribe during the initial subscribe.

Reply:
yes agree
i have updated the logic based on this.latestActivity now check since this value is get set once we get initial response of subscription, and is there throughout the subscription lifetime.